pageserver/src/tenant_mgr.rs

```diff
@@ -78,21 +78,23 @@ pub fn set_timeline_states(
             let walredo_mgr = PostgresRedoManager::new(conf, tenant_id);
 
             // Set up an object repository, for actual data storage.
-            match TenantConf::load(conf, tenant_id) {
-                Ok(tenant_conf) => {
-                    let repo: Arc<dyn Repository> = Arc::new(LayeredRepository::new(
-                        conf,
-                        tenant_conf,
-                        Arc::new(walredo_mgr),
-                        tenant_id,
-                        conf.remote_storage_config.is_some(),
-                    ));
-                    Tenant {
-                        state: TenantState::Idle,
-                        repo,
-                    }
+            let tenant_conf = match TenantConf::load(conf, tenant_id) {
+                Ok(tenant_conf) => tenant_conf,
+                Err(e) => {
+                    error!("Failed to load tenant state: {:?}", e);
+                    TenantConf::from(conf)
```

Comment:
Am I right that we will use default config if we failed to load the tenant specific one? If tenant has PITR of one year and we failed to load the config and loaded default one with PITR of e g one month GC will delete the data we actually need. 

I think we cannot use the default config here and we need to completely skip initialization of this tenant without affecting others. Plus ideally we should report this error somehow, maybe through timeline detail rest api response. But this code is undergoing through massive changes so this can be done separately to avoid conflicts

Reply:
Not sure if preventing access to some tenant data (because of config file corruption) is better solution than just using default config with risk of deleting data which should not be deleted.

But practically in this peace of code it is not so easy to implement what you have suggested (if not replace  `or_insert_with` method with something else).

Reply:
> Not sure if preventing access to some tenant data (because of config file corruption) is better solution than just using default config with risk of deleting data which should not be deleted.

I think it is better to be on the safe side (IMO data loss never should be considered as a viable option), other options:
* make this timeline readonly
* disable gc for it (can cause bloat without readonly option)
* retry whole operation (e g download)

Personally I'm inclined to retry the operation and deal with that in the same way as with failed download of a timeline. We probably need to guard this file with the checksum too.

@hlinnaka @SomeoneToIgnore thoughts?

> it is not so easy to implement what you have suggested

Feel free to leave a TODO assigned to me, because this code is being heavily reworked in #1163 so I can handle it there or in the follow up PR

Reply:
@knizhnik #1163 was merged, so this is no longer a problem. You can rebase your PR on top. Please reach out too me if you need help with that